### PR TITLE
add stop container

### PR DESCRIPTION
**Pull request by xlgao-zju**
Signed-off-by: Xianglin Gao <xlgao@zju.edu.cn>

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/frakti/22)

### CODE REVIEW

File: pkg/hyper/client.go [hunk 1]
```diff
@@ -189,6 +189,48 @@ func (c *Client) GetContainerInfo(container string) (*types.ContainerInfo, error
...
 
+// StopContainer stops a hyper container
+func (c *Client) StopContainer(containerID string, timeout int64) error {
+	//do checks about container and pod status
+	containerInfo, err := c.GetContainerInfo(containerID)
+	if err != nil {
+		return err
+	}
+
+	podInfo, err := c.GetPodInfo(containerInfo.PodID)
```
Comment (feiskyer) [inline pkg/hyper/client.go]: We don't really need this, checking `containerInfo.Status.Phase` is enough.

File: pkg/hyper/client.go [hunk 2]
```diff
@@ -189,6 +189,48 @@ func (c *Client) GetContainerInfo(container string) (*types.ContainerInfo, error
...
+	}
+
+	if containerInfo.Status.Phase != "running" {
+		return fmt.Errorf("Container %s is not running.", containerID)
+	}
+
+	ch := make(chan error, 1)
+
+	go func(containerID string) {
+		ch <- nil
```
Comment (feiskyer) [inline pkg/hyper/client.go]: What is this for?

File: pkg/hyper/client.go [hunk 3]
```diff
@@ -189,6 +189,48 @@ func (c *Client) GetContainerInfo(container string) (*types.ContainerInfo, error
...
+		defer cancel()
+
+		_, err := c.client.ContainerStop(ctx, &types.ContainerStopRequest{ContainerID: containerID})
+		ch <- err
+	}(containerID)
+
+	<-ch
+
+	select {
+	case <-time.After(time.Duration(timeout) * time.Second):
```
Comment (feiskyer) [inline pkg/hyper/client.go]: Check whether `timeout == 0`

File: pkg/hyper/client.go (hunk 2 above)
Comment (xlgao-zju) [inline pkg/hyper/client.go]: Code will be blocked util we get something [here](https://github.com/kubernetes/frakti/pull/22/files#diff-f7ec4be94a67a866942db2ebd145d763R224).
So, the timeout will be more precise.

**Comment by xlgao-zju** [comment]
updated.

**Comment by feiskyer** [comment]
LGTM. Thanks.

**Pull request merged**